src/H5FDros3.h

```diff
@@ -84,26 +135,94 @@ typedef struct H5FD_ros3_fapl_t {
     char    secret_key[H5FD_ROS3_MAX_SECRET_KEY_LEN + 1];
 } H5FD_ros3_fapl_t;
 
+/**
+ * \struct H5FD_ros3_fapl_ext_t
+ * \brief Extended configuration structure for holding the configuration data
+ *        to the #H5FD_ROS3 driver via a File Access Property List.
+ *
+ * \details H5FD_ros_fapl_ext_t is a public structure that is used to pass
+ *          configuration data to the #H5FD_ROS3 driver via a File Access
+ *          Property List.
+ *
+ * \var H5FD_ros3_fapl_t H5FD_ros3_fapl_ext_t::fa
+ *      Configuration structure for H5Pset_fapl_ros3() / H5Pget_fapl_ros3().
+ *
+ * \var char H5FD_ros3_fapl_ext_t::token[H5FD_ROS3_MAX_SECRET_TOK_LEN + 1]
+ *      A string which specifies the session/security token.
+ *
+ */
+typedef struct H5FD_ros3_fapl_ext_t {
+    H5FD_ros3_fapl_t fa;
+    char             token[H5FD_ROS3_MAX_SECRET_TOK_LEN + 1];
+} H5FD_ros3_fapl_ext_t;
```

Comment:
Ah, I had mostly imagined this structure solely existing in the tools code. The issue I see with putting it into the public API here is that if any new functionality is added, it would be another API-breaking change to add it into this structure, so at that point we may as well have just added everything to the original H5FD_ros3_fapl_t structure. One other issue I see here and that I ran into elsewhere is that having large char buffer fields in API structures can make it difficult for applications to have these variables on the stack for certain platforms and usually causes warnings in example code if they are used as stack variables.

Reply:
I have moved the structure to `h5tools_utils.h`.